packages/frontend/src/pages/insight-page/components/insight-viewer/components/insight-change-history/change-history-list.tsx

```diff
@@ -88,16 +132,30 @@ export const ChangeHistoryList = ({ insightId, insightFullName }: ChangeHistoryP
 
   const changeHistory = data.insight.changeHistory;
 
-  if (changeHistory) {
-    console.log(`Changes: ${changeHistory.edges.length}`);
-  }
-
   return (
     <Box>
+      {fetchingRollBack && (
+        <Alert info={`Rolling Back to commit.\nPlease wait, this might take some time.`} mb="1rem" />
+      )}
+
+      {!canEdit && (
+        <Alert
+          warning={`You don't have permissions to publish changes to this Insight. So you can't roll back to a previous commit.`}
+          mb="1rem"
+        />
+      )}
```

Comment:
Catching issues when the user doesn't have permission to push to the Insight repository